### DISCUSSION TITLE
Using Silero VAD in browser

### DISCUSSION THREAD

**Discussion by rtm-1dyakonov** [snakers4/silero-vad#175] [Q&A] [answered]
Hello!

You've made a huge amount of work with your SILERO models. The very last release with new formats like Onnx is fascinating.
I am really interested in using Sileor_VAD in web app. Do you have any examples of implementing web app with Onnx format of your model? Like in JavaScript.

**Comment by snakers4** [comment]
Hi,

We do not have ready JS examples or apps, we do not have an in-house frontend dev team.

But I have heard positive vibes about the new ONNX web or how is it called (the one to supersede `onnx.js`) and I have heard reports of people running it in browser (at least people were asking about resampling, etc).

Please feel free to ask around  (and search) in discussions here and in the telegram chat. Some people reported being able to launch / use ONNX web in the browser.

**Reply by snakers4** [reply]
https://github.com/microsoft/onnxruntime/tree/master/js/web

**Reply by snakers4** [reply]
Also important caveats for your use case:

- 8 kHz and 16 kHz models ended up being 2 separate models
- Hence we could not make an ONNX model with an if (it sounds silly, but we spent a lot of time trying, there were some deeper cryptic errors and issue)
- The current ONNX model is for 16 kHz only
- Resampling from 32kHz or 48 kHz is easy
- Upsampling from 8kHz can be done naively as well, or just try several options

**Comment by snakers4** [comment, marked as answer]
https://github.com/ricky0123/vad

**Reply by MariasStory** [reply]
Why did you change to ISC License?

**Reply by hongbo-miao** [reply]
@MariasStory you asked wrong person. I think you need ask the author of https://github.com/ricky0123/vad at that repo.
This repo https://github.com/snakers4/silero-vad is MIT license.
